doc/CONTRIBUTING.md

```diff
@@ -70,6 +70,85 @@ or reported for next major release if BC is not possible.
 
 If you are not sure of what to do, don't hesitate to open an issue about your PR project.
 
+#### Composing the PR
+
+Nobody is perfect, and submitting perfect changes that don't need any tweaks is
+quite a rare instance. After a few iterations, your PR, which is a list of
+commits, could look like this:
+
+- replace tabs with spaces (who did this?)
+- add flux capacitor feature
+- add missing documentation (silly me)
+- add missing tests (I love TDD, but…)
+- fix typo in documentation
+- set default value as suggested
+
+Ideally, we should be able to revert any commit in your PR and not have to
+revert the others too. For instance, if we decide to revert your feature
+because it turns out to cause more problems than it solves, we should not have
+to revert the commit with the feature itself, then the commit with the
+documentation, then the commit with the tests, etc…
+
+Another thing to take into consideration is that in some hopefully rare
+instances, you will need to keep some commits separate, because they do not
+really relate to what you are doing, but the "other" thing you fixed, while
+unrelated, somehow prevents you from doing your PR, or is a minor thing you
+feel does not warrant a PR by itself, especially if this PR is conflicting with
```

Comment:
Big :-1: on that, you know I'll will. :-)

Yeah, on rare case it could happens, but it's not a reason to add it on contributing section.

Contributing contains rules to follow, not special case we can excuse (and we should not).

BTW, PR with multiple "justified" commit are dangerous because all people are not git masters and could do some.

Imagine two "justified" commits A and B. The contributor add a "A fix commit" and when he tries to rewrite the history, accidentally fixed up on the B commit. How to fix that? AFAIK, beside redo all the commits, I don't think you can.

So I maybe feels insistent, but PR should concern one and only one thing.

Adopting the simple one line rule "If this line does not concern your change goal, does not touch it.".